packages/storybook/src/stories/grid-i18n.stories.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import { XGrid } from '@material-ui/x-grid';
+import '../style/grid-stories.css';
+import { useData } from '../hooks/useData';
+
+export default {
+  title: 'X-Grid Tests/i18n',
+  component: XGrid,
+  parameters: {
+    options: { selectedPanel: 'storybook/storysource/panel' },
+    docs: {
+      page: null,
+    },
+  },
+};
+export const CustomLocaleText = () => {
```

Comment:
I don't think that we need to store for this. A demo in the documentation to iterate on it, and a smoke end-to-end test should be enough. I think that it's important we work on the documentation by default, only move to storybook when it's no longer enough. So far, stories are meant for edge cases we can't cover in the documentation. We have talked about extending for

- rendering end-to-end tests
- rendering custom visual regression tests.


```suggestion
```

Reply:
Ok, noted. I usually add the story first because I use the storybook as a dev env. I'll create a demo in the docs. I see that there is already a page for it so I will update it 😉 [(DataGrid Localization)](https://material-ui.com/components/data-grid/localization/#data-grid-localization)

Reply:
> we work on the documentation first

Well, I don't think we should do the mentioned workflow. 
1. We add issues in the docs due to an extra plugin, so the environment is not properly isolated. And it's babel and not TS...
2. We render multiple demos on one page so, the logs are going to mix between grids, then breakpoints hit... 
3. The storybook seems faster to reload, and it does not need to reload the full page. Also is there type checking in docs?
In demos, you write in TS and JS at the same time, will you transpile manually at each change?

IMHO, you cannot build high-quality enterprise products properly in the middle of a next js doc website. 
The purpose of the storybook is to facilitate the development of components, providing tools, isolation, and all the required environment to successfully develop components.

Reply:
There is probably a balance to find between the two workflows. I see use cases for both. My concern was, at its core, that **when it's possible**, it's much better to start with the documentation. Given that we have a standalone documentation page for the localization. I thought that it was viable. At the end of the day, pick what works best for you. However, I think that we should be aware that by not working on the documentation first, we also make it more likely that the DX in the documentation won't be as great as it could have been. We have seen this problem happen a. couple of times so far with the data grid, where we introduced regressions. We should be careful to not make the docs a second-class citizen. The documentation is very important (based on the survey we did). So I think that it should be handled at the same time with any new feature as we do it for tests.

1. I don't follow everything.
  - What extra plugin? We have been working like this for years and at a very large scale in the main repo. I can safely say that it's working great. The root of the issue we got with the babel plugin doing the memoization was with the logic of the data grid, the docs, only surface it. We will have to fix this root issue anyway for the component composable API, then we can add the babel plugin back.
  - The isolated environement can be harmful too. It making us blind to how the components are used in real life. We have seen this happen multiple times on the data grid. e.g. wrong resize position, leaking keyboard isolation.
  - Why does it matter that it's Babel, not TS? The plan is to use the core bundling strategy with the enterprise components, they diverge from the convention. TS is great for static analysis, there is no point in using it for bundling, as far as I understand it.
2. I agree that working in isolation also has its use cases. How often do we work with the logs enabled? 
3.
 - Agree, the full reload of the documentation page sucks. I have no idea why it happens on material-ui-x. It would be great to fix it, to get the same DX as in the main repository.
 - For the JavaScript demos, the current workflow is to only work on the TypeScript ones. Once you are done (e.g. after a first review), you run the script that outputs the JavaScript ones. There is also a watch command to do such but I have never used it.